src/contracts/fungible/data/outcoins.rs

```diff
@@ -175,3 +177,23 @@ impl FromStr for ConsealCoins {
         }
     }
 }
+
+impl Eq for OutpointCoins {}
```

Comment:
This will also require manual implementation of `PartialEq` making sure we comparing amounts such that `NaN == NaN` (automatic implementation is `NaN != NaN`)